### ISSUE TITLE
Ability to convert to a static site

### ISSUE THREAD

**Issue by nolanlawson** [mastodon/mastodon#17596]
### Pitch

It's well-known that many Mastodon instance owners take their site down because it's too expensive to run, they lose interest, etc. Right now, though, the choice is between a fully-working Mastodon instance and a down site.

It would be great to have an option to export a Mastodon instance to a static site. This would allow any public links to statuses and profiles to still work correctly, even if no new content could be added.

### Motivation

Static sites are much cheaper to host than a Mastodon instance. In many cases, it can be done for free.

Having a static export option would allow Mastodon servers to live on in an "archived" format, providing an attractive alternative to instance owners who would otherwise just let their site go offline entirely.

**Comment by nightpool** [comment]
can't you do this with wget's "mirror" feature? maybe it would be better for us to just add this to the documentation rather then reinventing the wheel?

**Comment by nolanlawson** [comment]
If there is a good way to do this with some kind of off-the-shelf tool, then I'd be happy to use that, yeah. Adding some documentation to increase the visibility would be great!

**Comment by nolanlawson** [comment]
Thinking through the `wget` option, it seems to me that it could potentially cause a large bill or get blocked by DoS/abuse protection, especially if it spiders images/videos from an S3 bucket (which is not strictly necessary). Running the tool on the server itself using `localhost` would bypass that problem, but I wonder if that's possible with `wget`.

In any case, if it's possible, documentation would be good!

**Comment by shleeable** [comment]
"Data exporting" similar to the twitter would be interesting. That creates a static page with all of the media in one place. It's very good.

**Comment by Gargron** [comment]
That's off-topic because we already have data exporting for individual users and this issue is about turning a whole public server static. I think it's a bit out of scope for the project though.

**Comment by bkil** [comment]
It would raise GDPR & related privacy concerns if your former users who **own** the displayed posts would like to edit or ask for them to be removed in a future point in time. How would you do that without a backend or recording information for authorization of some sort?

**Comment by trwnh** [comment]
> How would you do that without a backend or recording information for authorization of some sort?

manually? i mean, you're dealing with a static site at that point, it's no different than someone else putting up an html page right now.
